### DISCUSSION TITLE
Enable diagnostics

### DISCUSSION THREAD

**Discussion by derjoerg** [kingsleyadam/local-abbfreeathome-hass#20] [Q&A] [answered]
Hi,

thinking about your new action to dump the configuration:
What about adding this to a diagnostics.py file, making this more HA "compliant"?

**Comment by kingsleyadam** [comment]
I like this, didn't even know this was an option. Looks like you can also redact data using `from homeassistant.components.diagnostics import async_redact_data` to remove any sensitive information and make it sharable.

**Reply by derjoerg** [reply]
Exactly, I just find it a bit problematic to get official documentation regarding that feature (at least I didn't find anything on developers).
But a lot of integrations have this integrated, so e.g. https://github.com/dougiteixeira/proxmoxve/blob/main/custom_components/proxmoxve/diagnostics.py might be interessting.

**Reply by kingsleyadam** [reply, marked as answer]
Yep, I was able to find a number of integrations that have this implemented. It's more straight forward than the service call I implemented. It also includes some good information about the HA instance itself as well (like confiirming which version of the integration they're running.

I'll back out the service call and implement this. Then update the README.md to add a troubleshooting section which includes providing this diagnostics download and how to enable more verbose logging for this integration.

**Comment by kingsleyadam** [comment]
https://github.com/kingsleyadam/local-abbfreeathome-hass/pull/22, I'll wait to merge this in until I can get the documentation updated. Maybe this weekend. Until then we can use the service call.
